apps/examples/board_specific_demo/ameba_ss_write_protection_demo/ss_write_protection_test.c

```diff
@@ -214,12 +259,51 @@ int sstorage_main(int argc, char *argv[])
 	}
 
 	printf("Read Complete\n");
+	sleep(1);
 
 	free(data.data);
 
 	printf("Start Security deinit!!\n");
 	(void)security_deinit(hnd);
 
+#ifdef CONFIG_MTD_BYTE_WRITE
+	/* Write data to Secure Storage flash directly */
+	printf("\nStart Flash Raw data write!!\n");
+	
+	if (slot_index == 0) {
+		address = SLOT0_START_ADDR;
+	} else {
+		address = SLOT1_START_ADDR + (SLOT_SIZE * (slot_index - 1));
+	}
+
+	printf("Write Raw data in Slot: %d, Area A\n", slot_index);
+	printf("Data will be written in SS if write protection is disabled.\n");
+	if (flash_write(address, length, (uint8_t *)rwbuf) < 0) {
+		printf("flash_write Failed. %d\n", __LINE__);
+	}
+
+	printf("Read Raw data after flash write\n");
+	printf("Read Raw data from Slot: %d, Area A\n", slot_index);
+	if (!rtl_ss_flash_read(address, length, (uint8_t *)rwbuf, 1)) {
+		printf("rtl_ss_flash_read() Fail, %d\n", __LINE__);
+	}
+
+	sleep(1);
+	address += SLOT_AREA_OFFEST;
+	printf("\nRead Raw data from Slot: %d, Area B\n", slot_index);
+	if (!rtl_ss_flash_read(address, length, (uint8_t *)rwbuf, 1)) {
+		printf("rtl_ss_flash_read() Fail, %d\n", __LINE__);
+	}
+
+	printf("Read Complete\n");
```

Comment:
Both Fail and Read can be printed.

how about change reture -1?

Reply:
Thank you for your review. I added function for read/write flash, and add "goto exit" if function failed.